### PR TITLE
Replace LONG DOUBLE with wide decimal for aggregates

### PR DESCRIPTION
**Pull request by mariadb-LeonidFedorov**
sum and avg used to have long double as underlying type. Here we replace it wide decimal

### CODE REVIEW

File: dbcon/mysql/ha_mcs_execplan.cpp [hunk 1]
```diff
@@ -3725,9 +3725,7 @@ ArithmeticColumn* buildArithmeticColumn(
     const CalpontSystemCatalog::ColType& leftColType = pt->left()->data()->resultType();
     const CalpontSystemCatalog::ColType& rightColType = pt->right()->data()->resultType();
 
-    // Only tinker with the type if all columns involved are decimal
-    if (datatypes::isDecimalOperands(mysqlType.colDataType,
```
Comment (drrtuy) [inline dbcon/mysql/ha_mcs_execplan.cpp:3729]: datatypes::isDecimalOperands should be removed with this change. It was its only call.

File: utils/rowgroup/rowaggregation.cpp [hunk 2]
```diff
@@ -1405,6 +1408,18 @@ void RowAggregation::doSum(const Row& rowIn, int64_t colIn, int64_t colOut, int
...
+        if (LIKELY(!isNull(fRowGroupOut, fRow, colOut)))
+        {
+            long double valOut = fRow.getLongDoubleField(colOut);
+            fRow.setLongDoubleField(valIn+valOut, colOut);
+        }
+        else
+        {
+            fRow.setLongDoubleField(valIn, colOut);
+        }
     } // end-of isWideDataType block
```
Comment (drrtuy) [inline utils/rowgroup/rowaggregation.cpp]: This comment becomes misguiding after the change.

File: utils/rowgroup/rowaggregation.cpp [hunk 3]
```diff
@@ -1379,19 +1379,22 @@ void RowAggregation::doSum(const Row& rowIn, int64_t colIn, int64_t colOut, int
             break;
         }
     }
-    if (LIKELY(!isWideDataType))
+
+    if (datatypes::hasUnderlyingWideDecimalForSumAndAvg(colDataType) && !isWideDataType)
```
Comment (drrtuy) [inline utils/rowgroup/rowaggregation.cpp]: Here is how I would optimize the block:
`    if (datatypes::hasUnderlyingWideDecimalForSumAndAvg(colDataType) || isWideDataType)
    {
        if (LIKELY(!isNull(fRowGroupOut, fRow, colOut)))
        {
            int128_t *valOutPtr = fRow.getBinaryField<int128_t>(colOut);
            int128_t sum = (isWideDataType) ? *valOutPtr + *reinterpret_cast<int128_t*>(wideValInPtr) : *valOutPtr + valIn;
            fRow.setBinaryField(&sum, colOut);
        }
        else
        {
            int128_t sum = valIn;
            fRow.setBinaryField(&sum, colOut);
        }
    }
    else
    {
        if (LIKELY(!isNull(fRowGroupOut, fRow, colOut)))
        {
            long double valOut = fRow.getLongDoubleField(colOut);
            fRow.setLongDoubleField(valIn+valOut, colOut);
        }
        else
        {
            fRow.setLongDoubleField(valIn, colOut);
        }
    } // end-of long double processing block`

Comment (drrtuy) [reply utils/rowgroup/rowaggregation.cpp]: I will send the snippet over Slack.

**Review by drrtuy — CHANGES_REQUESTED** [review]

File: dbcon/mysql/ha_mcs_execplan.cpp (hunk 1 above)
Comment (mariadb-LeonidFedorov) [reply dbcon/mysql/ha_mcs_execplan.cpp:3729]: ++

File: utils/rowgroup/rowaggregation.cpp (hunk 2 above)
Comment (mariadb-LeonidFedorov) [reply utils/rowgroup/rowaggregation.cpp]: ++

File: utils/rowgroup/rowaggregation.cpp (hunk 3 above)
Comment (mariadb-LeonidFedorov) [reply utils/rowgroup/rowaggregation.cpp]: ++

**Review by drrtuy — APPROVED** [review]

**Pull request merged**